bootstrap/kubeadm/internal/cloudinit/controlplane_init.go

```diff
@@ -62,6 +63,9 @@ func NewInitControlPlane(input *ControlPlaneInput) ([]byte, error) {
 	input.WriteFiles = input.Certificates.AsFiles()
 	input.WriteFiles = append(input.WriteFiles, input.AdditionalFiles...)
 	input.SentinelFileCommand = sentinelFileCommand
+	if input.AdminPass != "" {
+		input.ResetPwd = "- '(echo " + input.AdminPass + ";sleep 1;echo " + input.AdminPass + ") | passwd root' "
```

Comment:
I am not expert in this area
not sure this is the right way to input? 
do we have any reference of the usage here?

Reply:
If you mean the script:[(https://linuxhandbook.com/passwd-command/;)]
Of course, there are different ways to implement it through cloudinit, but I think it's enough to execute this script once by runcmd [https://github.com/madorn/cloud-init-guide/blob/master/13-run_commands.txt](url)